### PR TITLE
Changes required to build cppast with conan

### PR DESCRIPTION
**Pull request by Manu343726**
This PR adds all the changes I had to do to build cppast using conan to get the dependencies. Basically all dependencies are searched first through find_package() calls (With fallbacks to the old submodule/whatever way).

I'm not sure if you would like the way I search clang through, I'm open for changes and suggestions.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -15,6 +15,7 @@ endif()
 option(CPPAST_BUILD_TEST "whether or not to build the tests" ${CPPAST_IS_MAIN_PROJECT})
 option(CPPAST_BUILD_EXAMPLE "whether or not to build the examples" ${CPPAST_IS_MAIN_PROJECT})
 option(CPPAST_BUILD_TOOL "whether or not to build the tool" ${CPPAST_IS_MAIN_PROJECT})
+option(CPPAST_USE_OLD_LIBSTDCPP_ABI "Compile for the pre-C++11 libstdc++ ABI" OFF)
```
Comment (foonathan) [inline CMakeLists.txt]: Why does cppast need to take care of it?

**Comment by foonathan** [comment]
Looks good so far, I'm just curious about the arbitrary flag and ABI commit.

Also I need to take of foonathan/standardese#123 first.

**Comment by Manu343726** [comment]
About #23, for reference you can check what tinyrefl/tool/externals/externals.cmaje does when not using Conan (find_program(llvm-config) calls with  version checks etc)

**Comment by Manu343726** [comment]
(sorry, I'm on mobile)

Sorry, that arbitrary flags patch should not be there, it's something not related to Conan/dependencies. The reason I do that is to pass any flag (not just defines and include dirs) to the libclang parser. Some libs, Qt for example, require extra flags. I will rebase the branch asap to remove that commit from this PR

**Comment by Manu343726** [comment]
The ABI flag is to be able to build on environments using the old gcc ABI with a modern GCC. Conan models that case as part of its os-arch-compiler-stdlib-compilerversion build matrix

**Comment by Manu343726** [comment]
See https://docs.conan.io/en/latest/howtos/manage_gcc_abi.html?highlight=libcxx

**Comment by foonathan** [comment]
But why can you not do the ABI thing using `CMAKE_CXX_FLAGS`?

**Comment by foonathan** [comment]
Any update?

**Comment by Manu343726** [comment]
Sorry for the delay:

> But why can you not do the ABI thing using CMAKE_CXX_FLAGS?

I think you're right, I should be able to manage this from the conan side, passing the define explicitly through `CMAKE_CXX_FLAGS`.

> Any update?

I've not patched anything in my fork in the last two months, so if I fix that couple of things you noted this should be ready to merge.

**Comment by foonathan** [comment]
Hey, what's the current status? :D

**Comment by kmaragon** [comment]
Hi @Manu343726 I am also wondering about this. Or more than this, I hope all is ok. I've been personally benefiting immensely from all of your fantastic contributions to the C++ community over the years particularly around ctti and this kind of work. I'd love to see this get through the finish line. It seems like it was pretty much ready to go by the end here.

**Comment by foonathan** [comment]
I am assuming this has become obsolete according to something you've tweeted?

**Comment by Manu343726** [comment]
The reason why this PR stagnated was that first we have to figure out how to include libclang into mainstream conan first. There's a work in progress PR in conan-center-index, but it will take some time.

The changes needed in cppast were just about making dependency handling non intrusive, using cmake `find_package` is the best alternative. The reason why I haven't finished this PR is that I'm not sure what will be the final naming of the libclang package.

Once conan has an official llvm/libclang package I will push this PR again.

**Comment by foonathan** [comment]
You let me know when you want this merged, right?

**Comment by Manu343726** [comment]
Of course, don't worry, we just need to figure out llvm first. I'm just
patching stuff as bugs appear.

**Comment by saraedum** [comment]
Closing this one since nobody worked on this for years. Feel free to reopen.

**Pull request closed without merging**